xarray/backends/zarr.py

```diff
@@ -656,4 +656,4 @@ def maybe_chunk(name, var, chunks):
             return var
 
     variables = {k: maybe_chunk(k, v, chunks) for k, v in ds.variables.items()}
-    return ds._replace_vars_and_dims(variables)
+    return ds._replace_with_new_dims(variables)
```

Comment:
can this just be `_replace`? I find it confusing to "replace with new dims" but not specify "new dims".

Reply:
Hmmm, I think in this case `dims` need to be recomputed, since `variables` changed?